### PR TITLE
FIX-#2684: Add method level docs for Modin XGBoost

### LINKED ISSUE #2684: Add missed method level documentation for Modin XGBoost
### Describe the problem

Add missed method level documentation for Modin XGBoost

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2684

### CODE REVIEW

File: docs/modin_xgboost.rst [hunk 1]
```diff
@@ -24,23 +24,28 @@ XGBoost Train and Predict
...
-``predict`` is separate function unlike ``xgboost.Booster.predict`` which uses an additional argument
-``model``. ``model`` could be ``xgboost.Booster`` or output of ``modin.experimental.xgboost`` function.
+.. automodule:: modin.experimental.xgboost
+  :noindex:
+  :members: predict
 
-Both functions have additional parameters ``nthread`` and ``evenly_data_distribution``.
-``nthread`` sets number of threads to use per node in cluster.
-``evenly_data_distribution`` sets rule of distribution data between nodes in cluster.
+``predict`` is separate function unlike ``xgboost.Booster.predict`` which uses an additional argument
```
Comment (devin-petersohn) [inline docs/modin_xgboost.rst:37]:
```suggestion
``predict`` is separate function similar to ``xgboost.Booster.predict`` with an additional argument
```

File: docs/modin_xgboost.rst [hunk 2]
```diff
@@ -24,23 +24,28 @@ XGBoost Train and Predict
 Distributed XGBoost functionality is placed in ``modin.experimental.xgboost`` module.
 ``modin.experimental.xgboost`` provides a xgboost-like API for ``train`` and ``predict`` functions.
 
-``train`` has all arguments of ``xgboost.train`` function exclude the `evals_result`
+.. automodule:: modin.experimental.xgboost
+  :members: train
+
+As you can see ``train`` has all arguments of ``xgboost.train`` function exclude the `evals_result`
```
Comment (devin-petersohn) [inline docs/modin_xgboost.rst:30]:
```suggestion
``train`` has all arguments of ``xgboost.train`` function except for `evals_result`
```

**Review by devin-petersohn — CHANGES_REQUESTED** [review]

File: docs/modin_xgboost.rst (hunk 1 above)
Comment (prutskov) [reply docs/modin_xgboost.rst]: fixed

File: docs/modin_xgboost.rst (hunk 2 above)
Comment (prutskov) [reply docs/modin_xgboost.rst]: fixed

File: docs/modin_xgboost.rst [hunk 3]
```diff
@@ -24,23 +24,28 @@ XGBoost Train and Predict
...
-``predict`` is separate function unlike ``xgboost.Booster.predict`` which uses an additional argument
-``model``. ``model`` could be ``xgboost.Booster`` or output of ``modin.experimental.xgboost`` function.
+.. automodule:: modin.experimental.xgboost
+  :noindex:
+  :members: predict
 
-Both functions have additional parameters ``nthread`` and ``evenly_data_distribution``.
-``nthread`` sets number of threads to use per node in cluster.
-``evenly_data_distribution`` sets rule of distribution data between nodes in cluster.
+``predict`` is separate function similar to ``xgboost.Booster.predict`` with an additional argument
```
Comment (devin-petersohn) [inline docs/modin_xgboost.rst:37]:
```suggestion
``predict`` is similar to ``xgboost.Booster.predict`` with an additional argument, 
```

File: docs/modin_xgboost.rst [hunk 4]
```diff
@@ -24,23 +24,28 @@ XGBoost Train and Predict
 Distributed XGBoost functionality is placed in ``modin.experimental.xgboost`` module.
 ``modin.experimental.xgboost`` provides a xgboost-like API for ``train`` and ``predict`` functions.
 
-``train`` has all arguments of ``xgboost.train`` function exclude the `evals_result`
+.. automodule:: modin.experimental.xgboost
+  :members: train
+
+``train`` has all arguments of ``xgboost.train`` function except for `evals_result`
```
Comment (devin-petersohn) [inline docs/modin_xgboost.rst:30]:
```suggestion
``train`` has all arguments of the ``xgboost.train`` function except for ``evals_result``
```

**Review by devin-petersohn — COMMENTED** [review]
Minor cleanup.

File: docs/modin_xgboost.rst (hunk 3 above)
Comment (prutskov) [reply docs/modin_xgboost.rst]: Suggestion was applied.

File: docs/modin_xgboost.rst (hunk 4 above)
Comment (prutskov) [reply docs/modin_xgboost.rst]: Suggestion was applied.

**Comment by YarShev** [comment]
Since we added `xgboost` to `requirements-dev.txt` and `environment-dev.yml` then `Install XGBoost on Modin` section of the doc page needs to be fixed.

**Comment by prutskov** [comment]
@YarShev But we added `xgboost` only as development  dependency . `Install XGBoost on Modin section` is used for installing Modin for simple usage, and in this case we should say about needing of `xgboost` installation. For development dependencies we have instruction in [contributing](https://modin.readthedocs.io/en/latest/contributing.html#development-dependencies) docpage. What fixes do you mean?

**Comment by YarShev** [comment]
I had thoughts on adding some notes of `xgb` to installation page. But I think now is not time to add it as dependency for simple Modin usage. So, we can leave it as it is now.

**Review by YarShev — APPROVED** [review]
~LGTM!~ xgboost needs to be added to reqs to build docs.

**Comment by prutskov** [comment]
> ~LGTM!~ xgboost needs to be added to reqs to build docs.

added

**Review by devin-petersohn — APPROVED** [review]
LGTM!

**Pull request merged**